core/src/test/java/org/elasticsearch/action/bulk/BulkWithUpdatesIT.java

```diff
@@ -419,8 +457,8 @@ public void testBulkUpdateDocAsUpsertWithParent() throws Exception {
         byte[] addParent = new BytesArray("{\"index\" : { \"_index\" : \"test\", \"_type\" : \"parent\", \"_id\" : \"parent1\"}}\n" +
                 "{\"field1\" : \"value1\"}\n").array();
 
-        byte[] addChild = new BytesArray("{ \"update\" : { \"_index\" : \"test\", \"_type\" : \"child\", \"_id\" : \"child1\", \"parent\" : \"parent1\"}}\n" +
-                "{\"doc\" : { \"field1\" : \"value1\"}, \"doc_as_upsert\" : \"true\"}\n").array();
+        byte[] addChild = new BytesArray("{ \"update\" : { \"_index\" : \"test\", \"_type\" : \"child\", \"_id\" : \"child1\", " +
```

Comment:
I wonder if we should just build this with XContentBuilder? Or write it as formatted JSON?